openparticle/src.py

```diff
@@ -850,14 +895,23 @@ def _operate_on_state(self, other) -> "Fock":
         )
 
 
-class NumberOperator(ParticleOperator):
-    def __init__(self, particle_type, mode):
+class OccupationOperator(ParticleOperator):
+    def __init__(self, particle_type, mode, power):
         self.particle_type = particle_type
         self.mode = int(mode)
-        super().__init__(
-            self.particle_type
-            + str(self.mode)
-            + "^ "
-            + self.particle_type
-            + str(self.mode),
-        )
+        self.power = int(power)
+
+        if (self.particle_type != "a") and (power != 1):
+            raise ValueError(
+                "power of OccupationOperator can only be greater than 1 for bosonic ladder operators"
+            )
+
+        creation_op = ParticleOperator(self.particle_type + str(self.mode) + "^")
+        annihilation_op = ParticleOperator(self.particle_type + str(self.mode))
+        occupation_op = (creation_op**power) * (annihilation_op**power)
+        super().__init__(occupation_op.op_dict)
+
+
+class NumberOperator(OccupationOperator):
```

Comment:
Ah, disregard the top comment. I see now that we keep `NumberOperator` but it is rather a child class of `OccOp.`